code/game/gamemodes/changeling/changeling_powers.dm

```diff
@@ -23,6 +23,63 @@
 				src.verbs += P.verbpath
 
 	mind.changeling.absorbed_dna |= dna
+	var/mob/living/carbon/human/H = src
+	if(istype(H))
+		mind.changeling.absorbed_species |= H.species.name
+	for(var/language in languages)
+		mind.changeling.absorbed_languages |= language
+	return 1
+
+//Used to dump the languages from the changeling datum into the actual mob.
+/mob/proc/changeling_update_languages(var/updated_languages)
+
+	languages = list()
+	for(var/language in updated_languages)
+		languages += language

```

Comment:
If I understand it right, shouldn't this add_language(language.name) in the same standard as you set before?